### PR TITLE
Large, clumsy, unsorted pull request.

### PR DESCRIPTION
**Pull request by MarkMaldaba**
Hi Steve,

Here is a pull request with all my Behationary updates so far.

If you need me to split it up into separate pull requests, then I can do that, but it'll be a little while before I get round to it.  Thought I'd send the full request in the meantime.  The individual commit comments are pretty detailed where necessary, so hopefully it is not too hard to follow.

Headlines:
- Config file handling improved
- Possibility to specify local Behat installation
- Test code made optional
- A small amount of UI tidying
- Some code tidying

Feel free to reject if you want it broken up.

-- Mark

### CODE REVIEW

**Comment by MarkMaldaba** [comment]
I've just pushed another whole bunch of commits into this pull request, the main headline being that Behationary now supports multiple projects in nearly all respects.  The only exception being the quite important one; that it is not currently possible to define more than one project at a time!

The final bit of work will be to update the Config class to allow for multiple projects to be defined.

One method would be to look for multiple config.php files, but actually I think a much better method would be to switch to a YAML format for the config, which can allow for multiple projects in a single file, and also for project inheritance (to simplify re-use).

**Comment by meadsteve** [comment]
@MarkMaldaba sorry I missed this PR earlier. This isn't a project I'm actively working on at the moment but your changes all seem reasonable so happy to merge!
